tf2onnx/graph.py

```diff
@@ -1207,43 +1251,92 @@ def insert_new_node_on_output(self, op_type, output_name, name, domain=None, **k
         new_output = port_name(name)
         new_node = self.make_node(op_type, [output_name], attr=kwargs, outputs=[new_output], name=name, domain=domain)
 
-        to_replace = [n for n in self.get_nodes() if n != new_node]
+        to_replace = [self.get_node_by_name(n) for n in self._input_to_node_name[output_name]]
+        to_replace = [n for n in to_replace if n != new_node]
         self.replace_all_inputs(to_replace, output_name, new_output)
         return new_node
 
     def find_output_consumers(self, output_name):
         """Find all nodes consuming a given output."""
+        if output_name in self._input_to_node_name:
+            ops = self._input_to_node_name[output_name]
+            ops = [self.get_node_by_name(n) for n in ops]
+        else:
+            ops = self.get_nodes()
```

Comment:
Why would this case trigger?  Shouldn't the index always be up to date?  If output_name is not in self._input_to_node_name, ops should be [], I think.

Reply:
Done.